plugin/vcs/gitlab/gitlab.go

```diff
@@ -241,18 +241,28 @@ func (p *Provider) ExchangeOAuthToken(ctx context.Context, instanceURL string, o
 	if err != nil {
 		return nil, fmt.Errorf("failed to read oauth response body, code %v, error: %v", resp.StatusCode, err)
 	}
+	defer resp.Body.Close()
 
-	oauthToken := &vcs.OAuthToken{}
-	if err := json.Unmarshal(body, oauthToken); err != nil {
+	oauthToken := struct {
```

Comment:
Shall we update api/oauth.go to include Error and ErrorDescription fields?

Reply:
The one in vcs is specific to GitLab and probably needs to be refactored after we have the GitHub integration. So I am OK either way.

Reply:
different platforms may have different error field

Reply:
But my feeling is putting Error and ErrorDescription is closer to the end state in that the GitLab specific fields should be in gitlab.go. The existing vcs.OAuhToken should be in gitlab.go as well since that's gitlab specific.

Reply:
Maybe we should define the request and response in gitlab, and then convert to vcs.OauthToken

Reply:
That would be ideal

Reply:
> Maybe we should define the request and response in gitlab, and then convert to vcs.OauthToken

Yes, I like this approach better. Feel free to take whatever approach you're comfortable with. We can refactor it later while integrating with GitHub and other platforms.

Reply:
done